torchbenchmark/score/compute_score.py

```diff
@@ -8,31 +8,147 @@
 import sys
 import os
 import yaml
+import importlib
 
 from tabulate import tabulate
+from pathlib import Path
+from collections import defaultdict
 
 SPEC_FILE_DEFAULT = "score.yml"
 TARGET_SCORE_DEFAULT = 1000
 
-def compute_score(config, data, fake_data=None):
-    target = config['target']
-    score = 0.0
+def get_benchmark_norms(data):
+    """
+    Helper function which gets the normalization values per benchmark
+    by going through the reference data file.
+
+    This reference data can be user provided or by default the first file
+    in the directory of data files is considered as the reference data file
+    """
+    benchmark_norms = {b['name']: b['stats']['mean'] for b in data['benchmarks']}
+    return benchmark_norms
+
+def get_weights_from_spec(spec):
+    """
+    Calculates the static benchmark weights by iterating the spec
+    file and constructs a b_weight dictionary with (key, value) pair
+    is (task, weight_for_benchmark_per_task)
+
+    The spec here can be user defined .yaml file or if no spec is defined,
+    the default score.yaml is used
```

Comment:
It sounds like you don't have to pass in spec, and then score.yml is used, but it looks like you have to pass in spec.

I think spec should be an optional input.

This function should probably not know about the yaml format.  Instead, it should accept the simplest python data structure it needs to do its job, which may be a dictionary with a certain structure.  The stuff about yaml files or json files or whatever can be left outside in the function that calls this.

I'm not sure what the default should be if no spec is input.  It could either come from reading a file or it could come from looking at the enums.  I lean toward the enums since we could then delete a file and not have to duplicate content.  What do you think?